src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/CodeFixesForProjects.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Refactoring.V2
+{
+    [Shared]
+    [Export(typeof(CodeFixesForProjects))]
+    public class CodeFixesForProjects
+    {
+        private readonly ConcurrentDictionary<string, IEnumerable<CodeFixProvider>> _codeFixCache = new ConcurrentDictionary<string, IEnumerable<CodeFixProvider>>();
+        private readonly IAssemblyLoader _assemblyLoader;
+
+        [ImportingConstructor]
+        public CodeFixesForProjects(IAssemblyLoader assemblyLoader)
+        {
+            _assemblyLoader = assemblyLoader;
+        }
+
+        public IEnumerable<CodeFixProvider> GetAllCodeFixesForProject(string projectId)
+        {
+            if (_codeFixCache.ContainsKey(projectId))
+                return _codeFixCache[projectId];
+            return Enumerable.Empty<CodeFixProvider>();
+        }
+
+        public void LoadFrom(string projectId, IEnumerable<string> AnalyzerPaths)
+        {
+            var codeFixes = AnalyzerPaths
```

Comment:
I believe this should just get analyzer file reference that you created and call analyzerFileReference.GetAssembly and do what this does

http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/CodeFixes/CodeFixService.ProjectCodeFixProvider.cs,33

..

we don't have contract saying analyzer and fix must be in different dlls or saying fix must be in dll that ends with CodeFixes.

Reply:
Copied logic from that provider. ICodeFixProviderFactory is internal and left that part out, will that be problem in some cases?